packages/components/src/FormField/FormField.css

```diff
@@ -144,81 +101,168 @@ select.formField {
-  --field--padding-bottom: var(--space-smallest);
-  --field--padding-left: var(--space-base);
-  --field--padding-right: var(--space-base);
-  bottom: auto;
-  z-index: var(--elevation-base);
+.textarea .input {
+  resize: vertical;
+}
 
-  height: auto;
-  color: var(--color-text--secondary);
-  font-size: calc(var(--base-unit) * 0.875);
+.small .input {
+  --field--padding-top: var(--space-small);
+  --field--padding-bottom: var(--space-small);
 }
 
-.miniLabel.small .label {
-  display: none;
+.select .input {
+  --field--padding-top: 0;
+  --field--padding-bottom: 0;
 }
 
-.miniLabel.large .label {
+.miniLabel.select:not(.small) .input {
   --field--padding-top: var(--space-small);
-  --field--padding-bottom: var(--space-smallest);
-  --field--padding-left: var(--space-large);
-  --field--padding-right: var(--space-large);
 }
 
-.miniLabel.large textarea.formField {
-  --field--padding-top: var(--space-larger);
+/**
+ * Label
+ **/
+.label {
+  position: absolute;
+  top: var(--field--placeholder-offset);
+  z-index: calc(var(--field-base-elevation) + 1);
+  width: 100%;
+  padding-left: var(--field--padding-left);
+  padding-right: var(--field--padding-right);
+  overflow: hidden;
+  color: var(--field--placeholder-color);
+  line-height: calc(var(--base-unit) * 1.25);
+  text-align: var(--field--textAlign);
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  transform: translateY(var(--field--placeholder-transform));
+  transition: all var(--timing-quick);
 }
 
-.miniLabel.textareaLabel .label {
-  background-color: var(--color-surface);
+.miniLabel .label {
+  z-index: calc(var(--field-base-elevation) - 1);
+  font-size: calc(var(--base-unit) * 0.875);
 }
 
-/**
- * The custom property --formField-maxLength is getting defined inside the
- * component.
- */
-.maxLength {
-  width: calc(
-    (var(--base-unit) * var(--formField-maxLength)) + var(--field--padding-left) +
-      var(--field--padding-right)
-  );
+.textarea .label {
+  top: var(--field--padding-top);
+  transform: translateY(0);
+}
+
+.textarea.miniLabel .label {
+  top: var(--space-smallest);
+}
+
+.small.miniLabel .label {
+  display: none;
+}
+
+.large.miniLabel .label {
+  top: var(--space-small);
 }
 
+/**
+ * Postfix
+ **/
 .postfix {
   position: absolute;
   top: 50%;
-  right: var(--field--padding-right);
-  color: var(--color-text--secondary);
-  pointer-events: none;
+  right: var(--space-base);
+  z-index: calc(var(--field-base-elevation) - 1);
   transform: translateY(-50%);
 }
+
+/**
+ * Affix
+ **/
+
+.affixIcon {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  margin: 0 calc(var(--field--padding-left) * -1) 0 0;
+  /* 
+   * disabling property-no-unknown here as this is a real property
+   * that stylelint just doesn't know about. We may be able to remove
+   * this when we update our stylelint to not care about IE11.
+   */
+  /* stylelint-disable-next-line property-no-unknown */
+  aspect-ratio: 1/1;
```

Comment:
Since this is unsupported on Safari/Safari-for-iOS, can we use an `@supports` to get the magic of this when we have access to it?